src/Entity/DfTools.php

```diff
@@ -1396,6 +1407,26 @@ public static function getMinVariantPrices($products, $includeTaxes, $currencies
         return $minPricesByProductId;
     }
 
+    public static function maybeAddVariantsFirstParent($products, $idLang, $idShop)
+    {
+        $parentIndices = array_keys(
+            array_filter($products, function ($product) {
+                return self::isParent($product);
+            })
+        );
+        if (empty($parentIndices)) {
+            return $products;
+        }
+        $firstParentIndex = $parentIndices[0];
```

Comment:
Quizá se puede hacer también como:
```suggestion
    $indice = null;
    
    foreach ($product_index as $i => $product) {
        if (self::isParent($product)) {
            $indice = $i;
            break;
        }
    }
```

Lo digo porque quizá así se envita recorrer toda la lista de productos que es lo que imagino que hace el `filter` y a nosotros solamente nos interesa el primer padre.

Reply:
Ciertamente! Me parece una idea estupenda!! :open_hands: